tools/docker/tests/test_flows.sh

```diff
@@ -138,23 +130,32 @@ test_higher_layer_data_payload() {
 }
 test_topology() {
     status "test topology query"
-    eval send_bml_command "bml_wfa_ca_controller \"DEV_SEND_1905,DestALid,$AL_MAC,MessageTypeValue,0x0002\"" $redirect    
+    eval send_bml_command "bml_wfa_ca_controller \"DEV_SEND_1905,DestALid,$AL_MAC_agent1,MessageTypeValue,0x0002\"" $redirect    
     dbg "Confirming topology query was received"
-    docker exec -it repeater sh -c 'grep -i -q "TOPOLOGY_QUERY_MESSAGE" /tmp/$USER/beerocks/logs/beerocks_agent.log'
+    docker exec -it repeater1 sh -c 'grep -i -q "TOPOLOGY_QUERY_MESSAGE" /tmp/$USER/beerocks/logs/beerocks_agent.log'
 }
 test_init() {
     status "test initialization"
 
-    eval ${scriptdir}/test_gw_repeater.sh -f -d 5 $redirect || {
+    eval ${scriptdir}/test_gw_repeater.sh -f -r "repeater1" -r "repeater2" -d 5 $redirect || {
         err "start GW+Repeater failed, abort"
         exit 1
     }
-    AL_MAC=$(docker exec -it gateway ${installdir}/bin/beerocks_cli -c bml_conn_map | grep IRE_BRIDGE | awk '{print $5}' | cut -d ',' -f 1)
-    RADIO_WLAN0_MAC=$(docker exec -it gateway ${installdir}/bin/beerocks_cli -c bml_conn_map | grep "RADIO: wlan0" | head -1 | awk '{print $4}' | cut -d ',' -f 1 | tr --delete :)    
-    RADIO_WLAN0_MAC="0x${RADIO_WLAN0_MAC}"     
+    AL_MAC_agent1=$(docker exec -it gateway ${installdir}/bin/beerocks_cli -c bml_conn_map | grep IRE_BRIDGE | head -1 | awk '{print $5}' | cut -d ',' -f 1)
```

Comment:
Since there are so many addresses to retrieve, it would be better to do the bml_conn_map once, dump it in a file, and then parse it to different variables.

Also, the grep/head/awk can be done in a single simple awk script.

Both comments are nitpicks. I'm planning to do some improvements of these scripts anyway, so I can do it then.

Reply:
I agree with you, I'll try this now.